Source/Dafny/Compilers/SinglePassCompiler.cs

```diff
@@ -965,6 +979,7 @@ public class TypeArgumentInstantiation {
     /// Furthermore, EmitDestructor also needs to work for anonymous destructors.
     /// </summary>
     protected abstract void EmitDestructor(string source, Formal dtor, int formalNonGhostIndex, DatatypeCtor ctor, List<Type> typeArgs, Type bvType, ConcreteSyntaxTree wr);
+    protected virtual bool TargetLambdaRestrictedToExpressions => false;
```

Comment:
```suggestion
    protected virtual bool TargetLambdasRestrictedToExpressions => false;
```
Otherwise easy to misinterpret this as mutable instance state (as in "the lambda I'm currently translating is restricted to expressions")